Update format-hex tests to include -TestCase parameter

MiaRomero opened the pull request:
Addressing issue #3382 'Use -TestCases parameter in FormatHex.Tests.ps1'.

* Replaced the 'foreach' with the -TestCases parameter in the test helper functions.

Commit:
update tests to include -TestCase paramater in It block

JamesWTruher commented on test/powershell/Modules/Microsoft.PowerShell.Utility/FormatHex.Tests.ps1:
```diff
@@ -14,115 +14,127 @@
 #>
 
-function RunInputObjectTestCase($testCase)
+function RunInputObjectTestCase($testCases)
```
I think that these functions should just be inlined into the appropriate context block. Having these functions only makes things a little harder to find. it allows the test case data be closer to the actual test.

lzybkr commented:
@MiaRomero - the assignee of a PR is the person responsible for merging to master - and only maintainers can do that merge. We assign issues to regular contributors, but not the PR.

Commit:
Move test functions to appropriate context blocks

mirichmo commented:
@JamesWTruher or @Francisco-Gamino - Do you guys have any other concerns or comments?

mirichmo commented:
@miaromero - We need to add you to the Microsoft organization. I just noticed that you are not a member yet.

iSazonov commented on test/powershell/Modules/Microsoft.PowerShell.Utility/FormatHex.Tests.ps1:
```diff
@@ -443,9 +378,27 @@ Describe "FormatHex" -tags "CI" {
+            $errorThrown | Should Not Be $null
+            $errorThrown.FullyQualifiedErrorId | Should Match $ExpectedFullyQualifiedErrorId
+
+            $output | Should Not Be $null
```
We should check $output.Count >=1 or =1  not $null

MiaRomero replied:
Changed.

iSazonov commented on test/powershell/Modules/Microsoft.PowerShell.Utility/FormatHex.Tests.ps1:
```diff
@@ -216,9 +95,16 @@ Describe "FormatHex" -tags "CI" {
+
+            $result = Format-Hex -InputObject $InputObject
+
+            $result | Should Not Be $null
```
We can remove  - next line make the check.

MiaRomero replied:
Thanks, removed in all mentioned instances.

iSazonov commented on test/powershell/Modules/Microsoft.PowerShell.Utility/FormatHex.Tests.ps1:
```diff
@@ -284,13 +170,25 @@ Describe "FormatHex" -tags "CI" {
+
+            $result = $InputObject | Format-Hex
+
+            $result | Should Not Be $null
```
We can remove  - next line make the check.

iSazonov commented on test/powershell/Modules/Microsoft.PowerShell.Utility/FormatHex.Tests.ps1:
```diff
@@ -343,9 +231,27 @@ Describe "FormatHex" -tags "CI" {
+                $result = Format-Hex -LiteralPath $Path
+            }
+
+            $result | Should Not Be $null
```
We can remove  - next line make the check.

iSazonov commented on test/powershell/Modules/Microsoft.PowerShell.Utility/FormatHex.Tests.ps1:
```diff
@@ -389,15 +295,23 @@ Describe "FormatHex" -tags "CI" {
+
+            $result = Format-Hex -InputObject 'hello' -Encoding $Encoding
+
+            $result | Should Not Be $null
```
We can remove  - next line make the check.

iSazonov commented on test/powershell/Modules/Microsoft.PowerShell.Utility/FormatHex.Tests.ps1:
```diff
@@ -415,14 +329,35 @@ Describe "FormatHex" -tags "CI" {
+                $thrownError = $_
+            }
+
+            $thrownError | Should Not Be $null
```
We can remove  - next line make the check.

iSazonov commented on test/powershell/Modules/Microsoft.PowerShell.Utility/FormatHex.Tests.ps1:
```diff
@@ -443,9 +378,27 @@ Describe "FormatHex" -tags "CI" {
+                $output = Format-Hex -LiteralPath $InvalidPath, $inputFile1 -ErrorVariable errorThrown -ErrorAction SilentlyContinue
+            }
+
+            $errorThrown | Should Not Be $null
```
We can remove  - next line make the check.

MiaRomero commented:
@JamesWTruher fixed the >1 issue.   The way it was before was not actually testing the contents of the second file so when I corrected it, those tests failed because the output was on two lines.  I changed the input text to avoid the hassle of having two lines of output in the second expected result (then had to create a file with longer text to test the buffer underrun).

JamesWTruher commented on test/powershell/Modules/Microsoft.PowerShell.Utility/FormatHex.Tests.ps1:
```diff
@@ -415,14 +327,34 @@ Describe "FormatHex" -tags "CI" {
 
     Context "Continues to Process Valid Paths" {
 
+        $skipTest = ([System.Management.Automation.Platform]::IsLinux -or [System.Management.Automation.Platform]::IsOSX)
```
I'd be happier if this was combined with the construct in 378
IsLinux -or IsOSX -or (-not $certProviderAvailable)
then 378 could just be:
`It "<Name>" -Skip:$skip`
same with the code below

Commit:
combine skipTest conditions

Commit:
move skipTest to Describe block

adityapatwardhan commented on test/powershell/Modules/Microsoft.PowerShell.Utility/FormatHex.Tests.ps1:
```diff
@@ -480,14 +428,15 @@ Describe "FormatHex" -tags "CI" {
-            $result[1].ToString() | Should be "00000010   72 65 20 74 65 78 74                             re text         "
+            $result.Count | Should Be 3
+            $result[0].ToString() | Should be "00000000   4E 6F 77 20 69 73 20 74 68 65 20 77 69 6E 74 65  Now is the winte"
+            $result[1].ToString() | Should be "00000010   72 20 6F 66 20 6F 75 72 20 64 69 73 63 6F 6E 74  r of our discont"
```
Should use MatchExactly like here: https://github.com/PowerShell/PowerShell/blob/7b8c6e7f548e5ab98e9d36d073e5af7d989c8f79/test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1#L499

adityapatwardhan commented on test/powershell/Modules/Microsoft.PowerShell.Utility/FormatHex.Tests.ps1:
```diff
@@ -480,14 +428,15 @@ Describe "FormatHex" -tags "CI" {
-            $result[0].ToString() | Should be "00000000   54 68 69 73 20 69 73 20 61 20 62 69 74 20 6D 6F  This is a bit mo"
-            $result[1].ToString() | Should be "00000010   72 65 20 74 65 78 74                             re text         "
+            $result.Count | Should Be 3
+            $result[0].ToString() | Should be "00000000   4E 6F 77 20 69 73 20 74 68 65 20 77 69 6E 74 65  Now is the winte"
```
Should use MatchExactly like here: https://github.com/PowerShell/PowerShell/blob/7b8c6e7f548e5ab98e9d36d073e5af7d989c8f79/test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1#L499

adityapatwardhan commented on test/powershell/Modules/Microsoft.PowerShell.Utility/FormatHex.Tests.ps1:
```diff
@@ -480,14 +428,15 @@ Describe "FormatHex" -tags "CI" {
+            $result.Count | Should Be 3
+            $result[0].ToString() | Should be "00000000   4E 6F 77 20 69 73 20 74 68 65 20 77 69 6E 74 65  Now is the winte"
+            $result[1].ToString() | Should be "00000010   72 20 6F 66 20 6F 75 72 20 64 69 73 63 6F 6E 74  r of our discont"
+            $result[2].ToString() | Should be "00000020   65 6E 74                                         ent             "
```
Should use MatchExactly like here: https://github.com/PowerShell/PowerShell/blob/7b8c6e7f548e5ab98e9d36d073e5af7d989c8f79/test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1#L499

MiaRomero commented on test/powershell/Modules/Microsoft.PowerShell.Utility/FormatHex.Tests.ps1:
```diff
@@ -284,13 +171,24 @@ Describe "FormatHex" -tags "CI" {
     }
 
-    Context "Path Paramater" {
+    Context "Path and LiteralPath Paramaters" {
```
Fixed

Commit:
correct typo, change to MatchExactly

MiaRomero commented:
Thanks @adityapatwardhan for the comments - fixed all the Match to MatchExactly.

adityapatwardhan reviewed (approved):
LGTM
